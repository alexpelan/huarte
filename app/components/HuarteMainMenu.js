import React from 'react';
import {
  StatusBar,
  ListView,
  Text,
} from 'react-native';

import SeasonList from './SeasonList';
import Statistics from './Statistics';
import About from './About';

import styles from '../styles/styles';

const HUARTE_MENU_DATASOURCE = [
  {
    displayName: 'Play',
    component: SeasonList,
  },
  {
    displayName: 'Statistics',
    component: Statistics,
  },
  {
    displayName: 'About',
    component: About,
  },
];

class HuarteMainMenu extends React.Component {
  constructor(props) {
    super(props);
    const dataSource = new ListView.DataSource({
      rowHasChanged: (row1, row2) => row1 !== row2,
    });

    this.state = {
      dataSource: dataSource.cloneWithRows(HUARTE_MENU_DATASOURCE),
    };
  }

  selectMenuItem = (menuItem) => {
    const store = this.props.store;
    this.props.navigator.push({
      title: menuItem.displayName,
      component: menuItem.component,
      passProps: {
        store,
      },
    });
  };

  renderMenuRow = menuItem => (
    <Text
      style={styles.listItem}
      onPress={() => this.selectMenuItem(menuItem)}
    >
      {menuItem.displayName}
    </Text>
  );

  render() {
    StatusBar.setBarStyle('default', true);
    return (
      <ListView
        dataSource={this.state.dataSource}
        renderRow={menuItem => this.renderMenuRow(menuItem)}
        automaticallyAdjustContentInsets={false} // ????? https://github.com/facebook/react-native/issues/721
        style={styles.listView}
      />
    );
  }
}

export default HuarteMainMenu;
