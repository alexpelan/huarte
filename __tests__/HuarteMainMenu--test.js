import 'react-native';
import React from 'react';
import HuarteMainMenu from '../app/components/HuarteMainMenu';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <HuarteMainMenu />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});