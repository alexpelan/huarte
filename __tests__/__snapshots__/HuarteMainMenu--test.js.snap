// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  dataSource={
    ListViewDataSource {
      "items": 3,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
  style={
    Object {
      "backgroundColor": "#0000af",
      "paddingTop": 70,
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "marginBottom": 8,
          "textAlign": "center",
        }
      }
    >
      Play
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "marginBottom": 8,
          "textAlign": "center",
        }
      }
    >
      Statistics
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "marginBottom": 8,
          "textAlign": "center",
        }
      }
    >
      About
    </Text>
  </View>
</RCTScrollView>
`;
